repos/system_upgrade/el7toel8/libraries/requirearchver.py

```diff
@@ -0,0 +1,130 @@
+    :type required: list, tuple
+    """
+    if not isinstance(required, (list, tuple)):
+        raise TypeError('Required architectures has to be a list or tuple: {}'.format(required))
+
+    accepted = set(('ppc64le', 's390x', 'aarch64', 'x86_64'))
+    unsupported = set(required).difference(accepted)
+    if unsupported:
+        api.current_logger().warn('Unsupported architecture specified: {}'.format(unsupported))
+
+    detected = platform.machine()
+    reason = 'architecture'
+    if detected not in required:
+        _log_and_raise(reason, required, detected)
+
+    _log_passed(reason, required, detected)
+
+
+def _require_minor_version(destination, required, detected):
+    """
+    Stop actor's execution if requirement on minor version is not met.
+
+    :param destination: 'source' or 'target'
+    :type destination: string
+    :param required: list or tuple of versions - integers or strings in the form '<integer>['+'|'>'|'-'|'<']'
+    :type required: list, tuple
+    :param detected: Detected version
+    :type detected: int
+    """
+
+    reason = "{}'s minor version".format(destination)
+
+    if not isinstance(required, (list, tuple)):
+        raise TypeError("Required {reason} has to be a list or tuple: {required}".format(
+            reason=reason, required=required))
+
+    if all(isinstance(r, int) for r in required):
+        # required is list of versions
+        if detected not in required:
+            _log_and_raise(reason, required, detected)
+    else:
+        # required should be list of '<integer>['+'|'>','-','<']' strings
+        fun_map = {'+': operator.gt, '>': operator.gt, '-': operator.lt, '<': operator.lt}
+        try:
+            if not all(fun_map[r[-1]](detected, int(r[:-1])) for r in required):
+                _log_and_raise(reason, required, detected)
+        except (KeyError, ValueError, TypeError):
+            raise TypeError("Required {reason} has to be a list or tuple of integers or strings in the "
+                            "form '<integer>['+'|'>'|'-'|'<']': {required}".format(
+                                reason=reason, required=required))
+
+    _log_passed(reason, required, detected)
+
+
+# TODO: target's minor version will be specified using dynamic configuration phase, see #505 (add tests afterward)
+# def require_tgt_minor_version(required):
+#     """
+#     Stop actor's execution if requirement on minor version of target system is not met.
+#
+#     :param destination: 'source' or 'target'
+#     :type destination: string
+#     :param required: list or tuple of versions - integers or strings in the form '<integer>['+'|'>'|'-'|'<']'
+#     :type required: list, tuple
+#     :param detected: Detected version
+#     :type detected: int
+#     """
+#     detected = 0
+#     _require_minor_version('target', required, detected)
+
+
+def require_src_minor_version(required):
+    """
+    Stop actor's execution if requirement on minor version of source system is not met.
```

Comment:
I don't like this function. It is not doing what we expect to deliver. First, you target minor version only. But real code should operated with Major.Minor. So instead of 5, 6, 7... You should wirk with 7.5, 7.6, 7.7... 8.1, 8.3, 9.5....

Additionally, the logic is wrong. You are not stopping actor's execution. This is completely different story. It is expected to deliver functions that returns `True` or `False`. Because you need to process the actor maybe, but just don't want to process specific part of code or do various actions based on what upgrade path is processed.